src/App.js

```diff
@@ -0,0 +1,43 @@
+import React, { useState, useEffect } from "react";
+import HomePage from "./pages/HomePage/HomePage";
+import { ThemeProvider, createMuiTheme } from "@material-ui/core/styles";
+import "./App.scss";
+import Typography from "@material-ui/core/Typography";
+
+const THEME = createMuiTheme({
+  typography: {
+    h1: {
+      fontFamily: "Open Sans",
+      fontWeight: 700,
+      fontSize: 63,
+    },
+  },
+});
+
+function App() {
+  const [state, setState] = useState({ posts: [] });
+  useEffect(() => {
+    fetch("http://localhost:4000/posts")
+      .then((response) => response.json())
+      .then((posts) => setState({ posts: posts }));
+  });
```

Comment:
```suggestion
  useEffect(() => {
    fetch("http://localhost:4000/posts")
      .then((response) => response.json())
      .then((posts) => setState({ posts: posts }));
  }, [setState]);
```
Make sure to add the second argument when calling useEffect, so that fetching data process will be executed only when state is set initially.

Reply:
Do you have the resource to where you found this? just of out curriosity

Reply:
Read the section"Tip: Optimizing Performance by Skipping Effects" on this page: https://reactjs.org/docs/hooks-effect.html

Reply:
shouldn't it be `[state]` instead of `[setState]`. Also isn't the idea to prevent the useEffect from being called when a different posts request hasn't been made?